Source/ACE.Server/Network/NetworkSession.cs

```diff
@@ -245,7 +245,19 @@ private void HandlePacket(ClientPacket packet)
             uint issacXor = packet.Header.HasFlag(PacketHeaderFlags.EncryptedChecksum) ? ConnectionData.IssacClient.GetOffset() : 0;
             if (!packet.VerifyChecksum(issacXor))
             {
-                log.WarnFormat("[{0}] Packet {1} has invalid checksum", session.LoggingIdentifier, packet.Header.Sequence);
+                if (issacXor != 0)
+                {
+                    issacXor = ConnectionData.IssacClient.GetOffset();
+                    log.WarnFormat("[{0}] Packet {1} has invalid checksum, trying the next offset", session.LoggingIdentifier, packet.Header.Sequence);
+
+                    // not sure why there was a missed offset increment
+                    // results in a valid checksum
+                    // however, the client is now in a sort of "can't move" state.  The monsters react so the communication is still working, right?  fartwhif
```

Comment:
It has come to my attention that a fragment is being lost, so client->server traffic dies because there's no mechanism for requesting that the missing fragment is retransmitted.  Also, the whole network layer needs to have support for multiple queues (Packet.Header.table) as to not confuse the client leading to the missing fragment.